lib/HeroBonus.h

```diff
@@ -1106,12 +1120,13 @@ class DLL_LINKAGE CreatureTerrainLimiter : public ILimiter //applies only to cre
 	}
 };
 
-class DLL_LINKAGE CreatureFactionLimiter : public ILimiter //applies only to creatures of given faction
+class DLL_LINKAGE CreatureLevelLimiter : public ILimiter //applies only to creatures of given faction
 {
 public:
-	TFaction faction;
-	CreatureFactionLimiter();
-	CreatureFactionLimiter(TFaction faction);
+	uint32_t minLevel;
+	uint32_t maxLevel;

```

Comment:
I suggest to use ui32 types as they don't depend on platform (?)

Reply:
Not really.
ui32 types are old, vcmi-specific define in Global.h
uint32_t are standard c++ types.

In fact, ui32 and such are now simply define for uint32_t and such.
IMO we should prefer uint32_t in new code and slowly remove ui32 types.

Don't see any reason to actively remove them though (e.g. via mass replace)

Reply:
@IvanSavenko agree.

Reply:
truth, I see those types a part of c++ standard

Reply:
@Nordsoft91 this is linux kernel types, not standard ones)

Reply:
found, that's what I talked about - they are platform dependent:
https://en.cppreference.com/w/cpp/types/integer

> (provided if and only if the implementation directly supports the type)